File: Minimum Number[ 23-April-2023]/Solution.py
#   >> PYTHON CODE <<

def minimumNumber(self, n, arr):
        def greatest_common_divisor(a,b):
            if(b==0): return a
            
            return greatest_common_divisor(b,a%b)
        
        gcd=0
        
        for val in arr:
            gcd=greatest_common_divisor(gcd,val)
        
        return gcd

File: Minimum Number[ 23-April-2023]/Solution.java
//   >> JAVA CODE <<


class Solution {
    public static int greatestCommonDivisor(int a,int b)
    {
        if(b==0) return a;
        
        return greatestCommonDivisor(b,a%b);
    }
    public static int minimumNumber(int n, int[] arr) {
        
        int gcd=0;
        
        for(int i=0;i<n;i++)
        {
            gcd=greatestCommonDivisor(gcd,arr[i]);
        }
        
            return gcd;
    }
}
